Library/Homebrew/formula.rb

```diff
@@ -405,11 +414,21 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |p|
+      p.basename.to_s.start_with?("HEAD")
+    end
+
+    head_prefixes.sort_by do |p|
+      Tab.for_keg(Keg.new(p)).source_modified_time
```

Comment:
Two remarks:

1. You can drop the intermediate `Keg.new` as `Tab.for_keg` also directly operates on `Pathname` instances. (This will also skip checks in `Keg.new`, but they shouldn't be necessary here.)

2. Please use `pn` instead of `p` when enumerating `Pathname` instances (here and in the code above). I'm not sure we've been consistent with this in the existing code, but `pn` makes it a bit clearer that it's a `Pathname` object.